### PR TITLE
Add section about `Annotated` in `Field` usage

### PR DESCRIPTION
**Pull request by Viicos**
## Change Summary

As the issue was reported a few times latety, I figured adding a section in the docs was maybe worth it.

## Related issue number

Related to https://github.com/pydantic/pydantic/issues/7306 and a bunch of others.

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @lig

### CODE REVIEW

**Comment by Viicos** [comment]
Please review

**Review by adriangb — APPROVED** [review]

**Comment by adriangb** [comment]
Great thank you!

**Comment by adriangb** [comment]
Please fix the tests/example and this will get merged

File: docs/usage/fields.md [hunk 1]
```diff
@@ -38,6 +38,21 @@ class User(BaseModel):
...
 
+## Using `Annotated`
+
+The `Field` function can also be used together with `Annotated`.
+
+```py
+from typing_extensions import Annotated
+from uuid import uuid4
+
+from pydantic import BaseModel, Field
```
Comment (hramezani) [inline docs/usage/fields.md:49]:
```suggestion
from uuid import uuid4

from pydantic import BaseModel, Field
from typing_extensions import Annotated
```

File: docs/usage/fields.md [hunk 2]
```diff
@@ -38,6 +38,21 @@ class User(BaseModel):
...
 
+## Using `Annotated`
+
+The `Field` function can also be used together with `Annotated`.
+
+```py
+from uuid import uuid4
+
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
```
Comment (hramezani) [inline docs/usage/fields.md:49]:
```suggestion
from uuid import uuid4

from typing_extensions import Annotated

from pydantic import BaseModel, Field
```

**Comment by hramezani** [comment]
Thanks @Viicos

**Pull request merged**